client/src/pages/sports-grid/grid.js

```diff
@@ -1,45 +1,61 @@
 'use client'
 
 import Table from 'react-bootstrap/Table';
-import TeamSelector from './team-selector';
-import PlayerInput from './player-input';
+import Spinner from 'react-bootstrap/Spinner';
+
 import React from 'react';
+import { useState } from 'react';
+
+import PlayerInput from './PlayerInput';
+import TeamLogo from './TeamLogo';
+
+import * as constants from '../../utils/constants';
+import { useFetch } from '../../utils/hooks';
 
 export default function Grid() {
+    const sport = 'NBA';
+
+    const dataUrl = constants.routes[sport];
+    const logoSrc = constants.logos[sport];
+
+    const [isLoading, setIsLoading] = useState(true);
+
+    const data = useFetch(dataUrl, setIsLoading);
```

Comment:
so what this is doing currently is that on every render, you will call `useFetch` and get the same data every time, which is not ideal. You probably only want to call this only once (presumably on mount) and save the data (probably as a state). 

Look up "react fetch data with useEffect" and you can get some examples. This way, you can also put your loading logic in there.